GitUI/UserControls/FileStatusItem.cs

```diff
@@ -0,0 +1,26 @@
+﻿using GitCommands;
+
+namespace GitUI.UserControls
+{
+    public sealed class FileStatusItem
+    {
+        public FileStatusItem(GitRevision firstRev, GitRevision secondRev, GitItemStatus item)
+        {
+            SecondRevision = secondRev;
+            FirstRevision = firstRev;
+            Item = item;
+        }
+
+        /// <summary>
+        /// Selected (current or B in diff)
+        /// </summary>
+        public GitRevision SecondRevision { get; }
+
+        /// <summary>
+        /// First (Parent or A in diff)
+        /// </summary>
+        public GitRevision FirstRevision { get; }
+        public GitItemStatus Item { get; }
+        public override string ToString() => Item.ToString();
```

Comment:
❗ Possible NRE.
We either need to validate and throw in the ctor, or be defensive here.

Tests? :D

Reply:
See below